improver_tests/acceptance/test_threshold.py

```diff
@@ -168,7 +170,37 @@ def test_collapse_realization(tmp_path):
         "--collapse-coord",
         "realization",
     ]
-    run_cli(args)
+    with pytest.warns(None) as record:
+        run_cli(args)
+    for msg in record:
+        assert "Blending masked data without spatial" not in str(msg.message)
+    acc.compare(output_path, kgo_path)
+
+
+def test_collapse_realization_masked_data(tmp_path):
+    """Test thresholding and collapsing realizations using non-spatial weights,
+    where the data being thresholded is masked. This will result in a warning being
+    raised as such a collapse has not been fully tested when not using spatial
+    weights."""
+    kgo_dir = acc.kgo_root() / "threshold/masked_collapse"
+    kgo_path = kgo_dir / "kgo.nc"
+    input_path = kgo_dir / "input.nc"
+    output_path = tmp_path / "output.nc"
+    args = [
+        input_path,
+        "--output",
+        output_path,
+        "--threshold-values",
+        "500",
```

Comment:
The `threshold_values` here is different to the value in `test_collapse_realization()` (the test on unmasked data), is this because of different input data?

Reply:
Yes, that's correct.  The `threshold_values` are values of the diagnostic (eg a temperature of 280 K, or a cloud base height of 500 m) that will be used to generate binary fields based on whether the diagnostic exceeds (1) or does not (0) exceed this threshold.